### ISSUE TITLE
`log_dict` is missing `prefix` parameter that is mentioned in docs

### ISSUE THREAD

**Issue by rivi** [PyTorchLightning/pytorch-lightning#6124]
## 🐛 Bug

[`MetricCollection` documentation](https://pytorch-lightning.readthedocs.io/en/1.2.0/extensions/metrics.html#metriccollection) mentions using `self.log_dict(self.train_metrics, on_step=True, on_epoch=False, prefix='train')`. The `prefix` parameter doesn't seem to be present in `log_dict` function header.

### Expected behavior

`prefix` is most likely usable in this context, so this feature should be implemented. If not - the documentation should be fixed.

### Environment

 - Lightning Version: 1.2.0

**Comment by SkafteNicki** [comment]
@carmocca are you in favor of adding the argument to `log_dict` or fix the documentation?

**Comment by carmocca** [comment]
Fix the documentation. Does `MetricCollection` have a mechanism to set a prefix? If it doesn't, then the argument should be added there

**Comment by jspaezp** [comment]
@carmocca I agree this would be a nice addition and the implementation seems easy ...

```
-def __init__(self, metrics: Union[List[Metric], Tuple[Metric], Dict[str, Metric]]):
+def __init__(self, metrics: Union[List[Metric], Tuple[Metric], Dict[str, Metric]], prefix: Union[Literal[None],str]=None):
         super().__init__()
         if isinstance(metrics, dict):
+            prefix = prefix if prefix is not None else ""
             # Check all values are metrics
             for name, metric in metrics.items():
                 if not isinstance(metric, Metric):
                     raise ValueError(
                         f"Value {metric} belonging to key {name}"
                         " is not an instance of `pl.metrics.Metric`"
                     )
-                self[name] = metric
+                self[prefix + name] = metric
```

possibly add a warning as well if a list (and not a dict) is passed so you know that it will get ignored.
let me know if you want the PR. (I could also update the docs)

Kindest wishes,
Sebastian

**Comment by SkafteNicki** [comment]
@jspaezp your solution looks very similar to the one I had in mind:
https://github.com/PyTorchLightning/pytorch-lightning/compare/master...SkafteNicki:fix_prefix_arg
Feel free to submit a PR but please wait a couple of days for an update that will come to the metrics package (cannot disclose exactly what right now) :]

**Comment by jspaezp** [comment]
Hello @SkafteNicki I like the idea of your implementation and I think it could add as well an optional argument to the `clone` method.

I am thinking something like ...

```python
...
def clone(self, prefix = None):
    cloned = super().clone()
    if prefix is not None:
        cloned.prefix = prefix
    return cloned
...
``` 

Since it would make it a lot easier to make clones of large collections (train and test metrics for instance ...)
let me know when you feel would be a good time to work on it (although i feel like you got it covered, thanks a lot)

**Comment by uwaisiqbal** [comment]
I'm just updating to the latest version of pytorch lightning and I ran into this issue also. Do we have an ETA for when it will be released?
